src/common/src/row/mod.rs

```diff
@@ -201,6 +201,32 @@ pub trait RowExt: Row {
     fn is_null_at(&self, index: usize) -> bool {
         self.datum_at(index).is_none()
     }
+
+    fn exact_size_iter(&self) -> impl ExactSizeIterator<Item = DatumRef<'_>> {
```

Comment:
Can we directly let `Row::iter` return `impl ExactSizeIterator`?

Wrapping it to a new iterator is not that good because we lose the capabilities like specialization on the iterators provided by the standard library.